lib/remote/configobjectutility.cpp

```diff
@@ -198,11 +200,15 @@ bool ConfigObjectUtility::CreateObject(const Type::Ptr& type, const String& full
 		return false;
 	}
 
+	// AtomicFile doesn't create not yet existing directories, so we have to do it by ourselves.
 	Utility::MkDirP(Utility::DirName(path), 0700);
 
-	std::ofstream fp(path.CStr(), std::ofstream::out | std::ostream::trunc);
-	fp << config;
-	fp.close();
+	AtomicFile::Write(path, 0644, config);
+
+	Defer removeConfigPath([&path]{
+		// Something went wrong so remove the object's config accordingly.
+		Utility::Remove(path);
+	});
```

Comment:
Minor detail: you could move the comment out of the lambda and make it say that it's to remove the file in all the error cases and that the defer is cancelled on success so that the file is persisted in this case.